backend-base/src/main/java/org/eclipse/che/incubator/workspace/telemetry/finder/DevWorkspaceFinderImpl.java

```diff
@@ -50,11 +55,19 @@ public GenericKubernetesResource findDevWorkspace(String devworkspaceId) {
           }
         }
       );
+    } catch (IOException e) {
+      LOG.warn("Failed to read namespace in {}", Config.KUBERNETES_NAMESPACE_PATH, e);
     } catch (Exception e) {
       LOG.warn("Failed to find devworkspace with id: {} in namespace: {}", devworkspaceId, clientNamespace, e);
     }
 
     return devworkspace.get();
   }
 
+  private String getCurrentNamespace(KubernetesClient client) throws IOException {
+    BufferedReader br = new BufferedReader(new FileReader(Config.KUBERNETES_NAMESPACE_PATH));
+    String namespace = br .readLine();
+    br.close();
+    return namespace.strip();
+  }
```

Comment:
got a bit lost with implementation. Why do we pass the client in param? also wondering if it is a reliable way of getting the namespace based on the `KUBERNETES_NAMESPACE_PATH` ? should not we close reader in finally?

Reply:
Oops, thank you for pointing that out.

As for whether getting the namespace based on the `KUBERNETES_NAMESPACE_PATH` is reliable, it seems to work well empirically, but are there cases where `/var/run/secrets/kubernetes.io/serviceaccount` is not mounted to the container?

I've updated the code so that there is a new approach as well as the file reading approach. The new approach is reading the `DEVWORKSPACE_NAMESPACE` environment variable set on the workspace  containers by the [DWO](https://github.com/devfile/devworkspace-operator/blob/7c18bf96693ecc41c5f34a092abac8654223ea4d/pkg/library/env/workspaceenv.go#L33)

![image](https://user-images.githubusercontent.com/83611742/170373164-24ff7b3c-3fbf-4f6d-9a84-b80d88879846.png)

Reply:
Looks good, thanks